src/modules/0.0.16/aws_secret/index.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  CreateSecretCommandInput,
+  DescribeSecretCommandInput,
+  GetSecretValueCommandInput,
+  SecretsManager,
+  PutSecretValueCommandInput,
+  UpdateSecretCommandInput,
+  paginateListSecrets,
+  SecretListEntry,
+  SecretsManagerClient,
+} from "@aws-sdk/client-secrets-manager";
+import {
+  AWS,
+  crudBuilder2,
+  paginateBuilder,
+} from "../../../services/aws_macros";
+import { Context, Crud2, Mapper2, Module2 } from "../../interfaces";
+import * as metadata from "./module.json";
+import { Secret } from "./entity/secret";
+
+async function createSecret(
+  client: SecretsManager,
+  input: CreateSecretCommandInput
+) {
+  const res = await client.createSecret(input);
+  if (res) {
+    const s: SecretListEntry = {
+      Name: res.Name,
+      Description: input.Description,
+    };
+    return s;
+  }
+  return undefined;
+}
```

Comment:
This could just be

```suggestion
const createSecret = crudBuilderFormat<SecretsManager, 'createSecret', { Name: string, Description: String } | undefined>(
  'createSecret',
  (input) => input,
  (res) => !!res ? { Name: res.Name, Description: res.Description, } : undefined,
);
```
async function createSecret(
  client: SecretsManager,
  input: CreateSecretCommandInput
) {
  const res = await client.createSecret(input);
  if (res) {
    const s: SecretListEntry = {
      Name: res.Name,
      Description: input.Description,
    };
    return s;
  }
  return undefined;
}
```

Reply:
The advantage of this form is you can let Typescript's type inference take control and not need to import most of the types from the AWS SDK library.

Reply:
Most of your functions could be rewritten with one of these macro functions, btw.